### ISSUE TITLE
If 'enter' is used to select from dropdown in multi-select, the focus returns to ui-select input, but if 'tab' is used to select, the focus moves out of ui-select

### ISSUE THREAD

**Issue by jgentes** [angular-ui/ui-select#1261]
This may be intended behavior, but I think the general expected user experience for a multi-select field is to start typing, find something, press tab, type something else, press tab, something else, etc.

It works great if using the 'enter' key to select, but tab (and mouse click for that matter) don't return the focus to the ui-select input field.

![Animation](https://i.gyazo.com/334b0a9bc1a3aa0712eceb1bfa156c41.gif)

I'd imagine I can listen for the tab keypress event and return the focus using jquery, but I'd like to avoid that if possible.

Thanks, hopefully I'm missing something!

**Comment by Darex1991** [comment]
I have the same problem.. ;/

**Comment by bboyle** [comment]
:+1: just had a client ask for exactly this feature! :)

**Comment by synthecypher** [comment]
+1

**Comment by bboyle** [comment]
I've put up a PR with a possible fix here: https://github.com/angular-ui/ui-select/pull/1284

Feedback welcome.

**Comment by jgentes** [comment]
Thanks for the patch - here's what I'm seeing:

[![https://gyazo.com/e46eb4df4ef09ec34e8c25a424584ebc](https://i.gyazo.com/e46eb4df4ef09ec34e8c25a424584ebc.gif)](https://gyazo.com/e46eb4df4ef09ec34e8c25a424584ebc)

Also it's worth noting the specific requirement of `close-on-select="false"`

**Comment by schan93** [comment]
@jgentes Hi, I'm getting the exact issue that you posted right above this comment when I use ui-select and I click "tab." I don't have `close-on-selected` defined anywhere, would you happen if there's any other reason for why this might be happening? I have ui-select version 1.3.0 if that helps.

**Comment by bboyle** [comment]
@jgentes I've looked at the behaviour you describe above. It looks like `close-on-select="false"` overrides the default `reset-search-input="true"`

I'm not sure if that should be fixed in the same patch or separately. @wesleycho thoughts?

**Comment by user378230** [comment]
With this patch is it still possible to tab to another control on the page?

**Comment by bboyle** [comment]
@user378230 yes, if you close the list of options first (use 'Esc' on the keyboard). Don't want this to be a keyboard trap so if you find any bugs or have better suggestions please share.

**Comment by user378230** [comment]
I'm still concerned about whether this should be default behaviour or not.

If you were using ui-select only for creating new tags then this would make sense when you want to create a bunch of tags quickly. But if you're selecting from a list then I feel like the user would be pressing tab to select and move on. I guess I'll see what the other devs think.

**Comment by jgentes** [comment]
@user378230 once you tab-to-select, the focus should return to the input box, then if you push tab again, it would move on to the next control. Simply try it in a github text field by typing '@' and selecting someone's name. That's how ui-select should behave.

**Comment by okyanusatlas** [comment]
Now , as for today, you can choose with "tab" and "enter", anyway to prevent this to happen?
